Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceRayTracingClosestHit.azsl

```diff
@@ -0,0 +1,172 @@
+    if (payload.m_offScreen && !RayTracingGlobalSrg::m_rayTraceFallbackData)
+    {
+        // the hit is offscreen and we're not raytracing fallback data, report this as a miss
+        payload.m_hitT = -1.0f;
+        return;
+    }
+
+    // convert hitCoords to screen space
+    hitCoords.y = -hitCoords.y;
+    hitCoords.xy = (hitCoords.xy * 0.5f + 0.5f) * dimensions;
+
+    // retrieve Material data for the hit
+    RayTracingMaterialSrg::MaterialInfo materialInfo = RayTracingMaterialSrg::m_materialInfo[NonUniformResourceIndex(InstanceIndex())];
+    TextureData textureData = GetHitTextureData(materialInfo, vertexData.m_uv);      
+    
+    float3 vertexNormalWS = normalize(mul(meshInfo.m_worldInvTranspose, float4(vertexData.m_normal, 0)).xyz);
+
+    float3 normalWS = float3(0.0f, 0.0f, 0.0f);    
+    if (materialInfo.m_textureFlags & TEXTURE_FLAG_NORMAL)
+    {           
+        float3x4 world34 = ObjectToWorld3x4();
+        float4x4 world44;
+        world44[0] = world34[0];
+        world44[1] = world34[1];
+        world44[2] = world34[2];
+        world44[3] = float4(0.0f, 0.0f, 0.0f, 1.0f);
+        
+        float3 tangentWS = float3(0.0f, 0.0f, 0.0f);
+        float3 bitangentWS = float3(0.0f, 0.0f, 0.0f);            
+        ConstructTBN(vertexData.m_normal, 
+                     vertexData.m_tangent, 
+                     world44, 
+                     (float3x3)meshInfo.m_worldInvTranspose, 
+                     normalWS, 
+                     tangentWS, 
+                     bitangentWS);
+            
+        normalWS = GetWorldSpaceNormal(textureData.m_normal.xy, normalWS, tangentWS, bitangentWS);
+    }
+    else
+    {
+        normalWS = vertexNormalWS;
+    }
+  
+    float3 hitToCamera = ViewSrg::m_worldPosition.xyz - positionWS;
+    float distanceToCamera = length(hitToCamera);
+    hitToCamera = normalize(hitToCamera);
+
+    if (!payload.m_offScreen)
+    {
+        // secondary raycast back to the camera position, this will determine if there is data available on-screen
+        RayDesc secondaryRay;
+        secondaryRay.Origin = positionWS + (hitToCamera * 0.1f);
+        secondaryRay.Direction = hitToCamera;
+        secondaryRay.TMin = 0.0f;
+        secondaryRay.TMax = distanceToCamera;
+        
+        PayloadData secondaryPayload = (PayloadData)0;
+        secondaryPayload.m_cameraTrace = true;
+        TraceRay(
+            RayTracingSceneSrg::m_scene,
+            RAY_FLAG_SKIP_CLOSEST_HIT_SHADER,
+            0xFF,
+            0,
+            1,
+            0,
+            secondaryRay,
+            secondaryPayload);
+
+        payload.m_obstructed = (secondaryPayload.m_hitT >= 0.0f);
+    }
+
+    if (payload.m_obstructed && !RayTracingGlobalSrg::m_rayTraceFallbackData)
+    {
+        // the hit is obstructed and we're not raytracing fallback data, report this as a miss
+        payload.m_hitT = -1.0f;
+        return;
+    }
+
+    if (payload.m_obstructed || payload.m_offScreen)
```

Comment:
Would be nice to add detailed comments here about what we are trying to do here.